src/tools/merssh/command.cpp

```diff
@@ -252,6 +269,14 @@ void Command::maybeDoCMakePathMapping()
         }
 
         QString data = QString::fromUtf8(reader.data());
+        if (it.fileName() == "CMakeCache.txt") {
+            QRegularExpression cmakeHomeRegexp("CMAKE_HOME_DIRECTORY:INTERNAL=(.*)");
+            QRegularExpressionMatch cmakeHomeMatch=cmakeHomeRegexp.match(data);
+            if (cmakeHomeMatch.hasMatch())
+                relativeRoot = rootFromDir(cmakeHomeMatch.captured(1));
+        }
+        if (!relativeRoot.isEmpty())
+            data.replace(relativeRoot, sharedTargetRoot);
```

Comment:
.cbp files on my machine contain paths with backslashes too `<Option virtualFolder="CMake Files\..\..\..\..\..\usr\lib\cmake\Qt5Core\"/>` - shouldn't these be replaced?

Reply:
To me it seems QtC does not use those

Reply:
I would consider replacing these too so that we do not need to ponder about in future.

Reply:
What would you replace them with? "/usr/lib/cmake/Qt5Core/" or "\usr\lib\cmake\Qt5Core\" or "CMake Files\usr\lib\cmake\Qt5Core" or "CMake Files/usr/lib/cmake/Qt5Core/" or somethine else?

Reply:
Hm, I think `sharedTargetRoot` should look more like the original `relativeRoot`, i.e. it should be a concatenation of source path and target root path relative to source path.

Better expressed by example:

```diff
-<Unit filename="/home/mersdk/share/tmp/foobar/../../../../../usr/lib/cmake/Qt5Core/Qt5CoreMacros.cmake">
+<Unit filename="/home/user/tmp/foobar/../../SailfishOS/mersdk/targets/SailfishOS-3.3.0.16-i486/usr/lib/cmake/Qt5Core/Qt5CoreMacros.cmake">
```

(Given that `/home/user/` is user home on host and the SDK is installed under `/home/user/SailfishOS`.)

This is then more strightforward to be extend to the `Option` tag:

```diff
-    <Option virtualFolder="CMake Files\..\..\..\..\..\usr\lib\cmake\Qt5Core\"/>
+    <Option virtualFolder="CMake Files\..\..\SailfishOS\mersdk\targets\SailfishOS-3.3.0.16-i486\usr\lib\cmake\Qt5Core\"/>
```

Reply:
And what if SDK is installed in C:\SailfishOS, user home directory is D:\Users\myuser and he is using alternative source directory in E:\src?

Reply:
The more I think about this, the more it seems to me that it doesn't make any sense to modify the virtualFolder at all. There are always cases when the value wouldn't make any sense, and as long as it is not used anywhere (QtC only checks that the value exists), it does not matter what it is. Modifying it is just wasted time.

Reply:
yeah, lovely :\